ambry-api/src/main/java/com.github.ambry/account/AclService.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.account;
+
+import com.github.ambry.rest.SecurityService;
+
+
+/**
+ * A service that handles authorization of requests. This will be provided to the {@link SecurityService} for making
+ * access decisions that might require the help of an external service. The call to
+ * {@link #hasAccess(Object, Resource, Operation)} should not require expensive network calls, since it will potentially
+ * be used on the critical path. Instead, the {@link AclService} should maintain a cache of ACLs for the resources it
```

Comment:
I think this should just say that `hasAccess()` should be quick since this may in the critical path. Caching etc is an impl detail.